File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/models/base/AppContext.java
package app.editors.manager.mvp.models.base;

public enum  AppContext {

    NONE(0),
    MY(1),
    SHARE(2),
    COMMON(3),
    PROJECTS(4),
    TRASH(5);

    private final int mValue;

    AppContext(int value) {
        mValue = value;
    }

    public int getValue() {
        return mValue;
    }

    public static AppContext  getEnum(int value){
        for (AppContext e : AppContext.values()) {
            if(e.getValue() == value)
                return e;
        }
        return AppContext.NONE;
    }

}

File: app_manager/core/src/main/java/app/documents/core/network/manager/models/base/AppContext.kt
package app.documents.core.network.manager.models.base

enum class AppContext(val value: Int) {
    NONE(0), MY(1), SHARE(2), COMMON(3), PROJECTS(4), TRASH(5);

    companion object {
        fun getEnum(value: Int): AppContext {
            for (e in values()) {
                if (e.value == value) return e
            }
            return NONE
        }
    }
}